internal/pkg/manifest/env.go

```diff
@@ -97,9 +97,12 @@ type EnvironmentConfig struct {
 	CDNConfig     EnvironmentCDNConfig     `yaml:"cdn,omitempty,flow"`
 }
 
-// IsIngressRestrictedToCDN returns whether or not an environment has its
+// ALBIngressRestrictedToCDN returns whether an environment has its
 // Public Load Balancer ingress restricted to a Content Delivery Network.
-func (mft *EnvironmentConfig) IsIngressRestrictedToCDN() bool {
+func (mft *EnvironmentConfig) ALBIngressRestrictedToCDN() bool {
+	if !mft.HTTPConfig.Public.Ingress.IsEmpty() {
+		return aws.BoolValue(mft.HTTPConfig.Public.Ingress.CDNIngress)
+	}
 	return aws.BoolValue(mft.HTTPConfig.Public.SecurityGroupConfig.Ingress.RestrictiveIngress.CDNIngress)
```

Comment:
We'll probably need some comments here:
```suggestion
        // Check the fixed manifest first. This would be `http.public.ingress.cdn`.
        // For more information, see https://github.com/aws/copilot-cli/pull/4068#issuecomment-1275080333
	if !mft.HTTPConfig.Public.Ingress.IsEmpty() {
		return aws.BoolValue(mft.HTTPConfig.Public.Ingress.CDNIngress)
	}
	// Fall through to the old manifest: `http.public.security_groups.ingress.cdn`.
	return aws.BoolValue(mft.HTTPConfig.Public.SecurityGroupConfig.Ingress.RestrictiveIngress.CDNIngress)
```